data_safe_haven/resources/workspace/ansible/host_vars/localhost.yaml

```diff
@@ -0,0 +1,122 @@
+---
+
+# Packages by category
+# Each category has 'common' packages for when the package name is identical
+# for all releases
+# Other packages are listed in dicts with the name of the release as given by
+# `ansible_facts['distribution_release']`
+package_categories:
+  - category: admin
+    common:
+      - auditd
+      - bats
+      - clamav
+    focal:
+    jammy:
+    noble:
+  - category: browsers
+    common:
+      - firefox
+    focal:
+    jammy:
+    noble:
+  - category: database
+    common:
+      - libpq-dev     # interact with PostgreSQL databases
+      - msodbcsql17   # interact with Microsoft SQL databases
+      - unixodbc-dev  # interact with Microsoft SQL databases
+    focal:
+    jammy:
+    noble:
+  - category: development
+    common:
```

Comment:
- Do we want to keep Java, scala? I would assume they are much less common
- rustc will be installed as part of cargo (we could add it explicitly though)
- Is spark-shell any use given we don't have a spark server?

Reply:
I think supporting as many programming languages as possible is part of our aim. Java and scala should be in Ubuntu's repos, so I imagine that's quite easy to support (`maven` has been a bit of a problem with scala projects in the past, but we can cross that bridge if we come to it again). Agreed that rustc should be installed by cargo, but let's be explicit. We can drop spark-shell if it's useless without a cluster (I don't know whether it can be run standalone).

Are there any other languages we should be considering here? e.g. dotnet/mono support?

Reply:
Agreed. I think in practice Python > R >> everything else, but it shouldn't be too difficult to support more.

Didn't we find dotnet had some daemon that wanted to phone home?

Reply:
We are missing functional languages, LISP dialects, Haskell, F#.

Reply:
F# would be covered by dotnet (I think - check with e.g. Evelina). If you can find a LISP/Haskell practitioner and find out what they need installed then yes, let's do that.

Reply:
> Didn't we find dotnet had some daemon that wanted to phone home?

I think you can (a) turn off dotnet telemetry and (b) ignore it, since failure to phone home doesn't break anything.

Maven was more of a problem, since scala programs wouldn't run until they'd updated all their packages.